backend/main/views_helper.py

```diff
@@ -3,6 +3,20 @@
 from backend.protzilla.steps import StepManager, Step
 from backend.protzilla.utilities import name_to_title
 
+
+def convert_name(name: str) -> [str, str]:
+    """
+    Converts a run name or database name to a valid filename by replacing spaces and special characters with underscores.
+    """
+    original_name = name
+    name = name.strip()
+    name = re.sub(r"[^\w\s-]", "_", name)  # replace special characters with underscores
+    name = re.sub(r"[-\s]+", "_", name)  # replace spaces and hyphens with underscores
```

Comment:
Find den Regex auch wirklich gut
Warum hast du dich dazu entscheiden, den in 2 Regexe aufzuteilen?

Ich würde in das ersetzen so wenig Logik, wie möglich einbauen, damit die nutzende Person möglichst wenig verwirrt ist von dem, was passiert. Daher fände ich es besser, wenn mehrere Lehrzeichen auch zu mehreren underscores umgewandelt werden. Normalerweise verwendet mensch glaub ich sowieso nicht mehrere Leerzeichen und wenn doch hats vielleicht einen Grund

Im Meeting hatte ich es so verstanden, dass wir `-` nicht in Underscores übersetzen wollen.  Bist du dir sicher, dass wir `-`nicht erlauben wollen?
```suggestion
    name = re.sub(r"[^\w-]", "_", name)  # replace special characters with underscores
```

Reply:
ich hab es jetzt noch mal umgeschrieben und jetzt auch in einer Zeile, - werden dabei nicht ersetzt, passt dir das so?